code/game/list.c

```diff
@@ -70,7 +70,7 @@ static int list_add_impl(list_p list, void* data, dataType_t type, size_t size,
     if (isPtr) {
         node->cont->data = malloc(size);
         if (node->cont->data == NULL) {
-            return 0;
+            goto on_error;
```

Comment:
I'm not a huge fan either to be honest, but it works as a finally block without adding much complication. I wouldn't really use them in any other context except for resource cleanup a la finally blocks.